### PR TITLE
Make scripts run from admin area

### PR DESCRIPTION
**Pull request by mike-lvov**
closes https://github.com/sparkletown/internal-sparkle-issues/issues/1641

### CODE REVIEW

File: src/settings/urlSettings.ts
Comment (goran-peoski-work) [inline src/settings/urlSettings.ts:67]: The naming convention is described in the file itself, lines 29 to 31

File: src/pages/SpacesDashboard/SpacesDashboard.tsx
Comment (goran-peoski-work) [inline src/pages/SpacesDashboard/SpacesDashboard.tsx:128]: If this is the only button (not regarding typing a URL in browser's address bar) it can be wrapped in `<TesterRestricted>`

File: src/pages/Admin/Tools/Tools.tsx
Comment (goran-peoski-work) [inline src/pages/Admin/Tools/Tools.tsx:59]: Can benefit from `useForm()` not necessarily required for this PR

File: src/pages/Admin/Tools/Tools.tsx
Comment (goran-peoski-work) [inline src/pages/Admin/Tools/Tools.tsx:40]:
```suggestion
```

File: functions/analytics.js
Comment (goran-peoski-work) [inline functions/analytics.js:46]:
```suggestion
```

File: src/pages/Admin/Tools/Tools.tsx
Comment (goran-peoski-work) [inline src/pages/Admin/Tools/Tools.tsx:35]:
```suggestion
```

File: src/pages/Admin/Tools/Tools.tsx
Comment (goran-peoski-work) [inline src/pages/Admin/Tools/Tools.tsx:17]: `<ButtonNG>` ?

File: src/components/organisms/AppRouter/AdminSubRouter.tsx
Comment (goran-peoski-work) [inline src/components/organisms/AppRouter/AdminSubRouter.tsx:54]: Other routes and with `_PARAM_URL` to denote their structure/purpose

File: src/components/organisms/AppRouter/AdminSubRouter.tsx
Comment (goran-peoski-work) [inline src/components/organisms/AppRouter/AdminSubRouter.tsx:17]: Maybe a good idea to rename it to `ToolsPage`.

Should make maintenance a bit easier if all top level components (directly under a route) end in `*Page`

File: src/pages/Admin/Tools/Tools.tsx
Comment (goran-peoski-work) [inline src/pages/Admin/Tools/Tools.tsx:34]: can benefit from `useAsyncFn`

File: src/pages/Admin/Tools/Tools.tsx
Comment (goran-peoski-work) [inline src/pages/Admin/Tools/Tools.tsx:60]: it will be much easier if we don't depend on dynamic forms but each script have it's own custom form, just like it has its own custom cloud function

File: functions/src/utils/time.js
Comment (goran-peoski-work) [inline functions/src/utils/time.js:1]: is this a copy-paste from `time.ts`?

File: functions/package.json
Comment (goran-peoski-work) [inline functions/package.json:16]: why is this necessary?

**Review by goran-peoski-work — COMMENTED** [review]
Good start. I've added few comments that might help improving the code (be it now or in future PR). Didn't spend time testing it, so will leave it up for some one else to approve it

File: functions/src/utils/time.js
Comment (mike-lvov) [reply functions/src/utils/time.js:1]: yes

File: src/pages/Admin/Tools/Tools.tsx
Comment (mike-lvov) [reply src/pages/Admin/Tools/Tools.tsx]: Not relevant with the latest commit

File: src/pages/Admin/Tools/Tools.tsx
Comment (mike-lvov) [reply src/pages/Admin/Tools/Tools.tsx]: Not relevant with the latest commit

File: functions/analytics.js
Comment (mike-lvov) [reply functions/analytics.js]: Not relevant with the latest commit

File: src/pages/Admin/Tools/Tools.tsx
Comment (mike-lvov) [reply src/pages/Admin/Tools/Tools.tsx]: Not relevant with the latest commit

File: src/pages/SpacesDashboard/SpacesDashboard.tsx
Comment (mike-lvov) [reply src/pages/SpacesDashboard/SpacesDashboard.tsx]: was wrapped in the https://github.com/sparkletown/sparkle/pull/2747/commits/48e69fc02ffb5e7c698db5b8a414d473df6eb3c8

File: src/settings/urlSettings.ts
Comment (mike-lvov) [reply src/settings/urlSettings.ts]: Fixed in https://github.com/sparkletown/sparkle/pull/2747/commits/73bc0106ad8f80a3da7b69b082166b60d78fc4e8

File: src/pages/Admin/Tools/Tools.tsx
Comment (mike-lvov) [reply src/pages/Admin/Tools/Tools.tsx]: Not sure what do you mean by this

File: src/components/organisms/AppRouter/AdminSubRouter.tsx
Comment (mike-lvov) [reply src/components/organisms/AppRouter/AdminSubRouter.tsx]: Fixed in https://github.com/sparkletown/sparkle/pull/2747/commits/73bc0106ad8f80a3da7b69b082166b60d78fc4e8

File: src/pages/Admin/Tools/Tools.tsx
Comment (mike-lvov) [reply src/pages/Admin/Tools/Tools.tsx]: Done in https://github.com/sparkletown/sparkle/pull/2747/commits/73bc0106ad8f80a3da7b69b082166b60d78fc4e8

File: functions/package.json
Comment (mike-lvov) [reply functions/package.json]: Goood catch! Removed in https://github.com/sparkletown/sparkle/pull/2747/commits/f98865aa73e97ffb903a45cab387020444385cc4

File: src/pages/Admin/Tools/Tools.tsx
Comment (mike-lvov) [reply src/pages/Admin/Tools/Tools.tsx]: Let's gather some more data about the use cases of these. If the evidence will prove your guess, we could easily change it later

File: functions/analytics.js
Comment (goran-peoski-work) [inline functions/analytics.js:1]: why is this needed? why write to disk?

File: src/pages/Admin/Tools/Tools.tsx
Comment (goran-peoski-work) [reply src/pages/Admin/Tools/Tools.tsx]: A more elaborate form (example, like in `<WorldGeneralForm/>`) that has `useForm` and `useAsyncFn` hooks to deal more robustly with the input. Not necessarily in scope for this PR, but maybe for a next one

File: functions/analytics.js
Comment (goran-peoski-work) [inline functions/analytics.js:226]: Is this used anywhere?

File: src/pages/Admin/ToolsPage/ToolsPage.scss
Comment (goran-peoski-work) [inline src/pages/Admin/ToolsPage/ToolsPage.scss:12]: If this is for a top level component, shouldn't it have it's own file in its own directory with own `index.ts`?... the works.

**Review by goran-peoski-work — APPROVED** [review]
Looks OK to me. Added few comments, can approve now, but would suggest get another round of prettying up the code and maybe another set of eyes before merge.

File: functions/analytics.js
Comment (mike-lvov) [reply functions/analytics.js:226]: Yep, right below. Note that I didn't write the script itself, just copied it

File: src/pages/Admin/ToolsPage/ToolsPage.scss
Comment (mike-lvov) [reply src/pages/Admin/ToolsPage/ToolsPage.scss]: Done in https://github.com/sparkletown/sparkle/pull/2747/commits/2949bdd7d241295eb345942957244b8dfe7f6634

**Pull request merged**
